intelmq/lib/BlueCoatAPI.py

```diff
@@ -0,0 +1,187 @@
+                self.logger.debug("Successfully parsed get_category_list()-response from BC-API.")
+
+                if 'Categories' in parsed_response:
+                    if 'Category' in parsed_response['Categories']:
+                        return parsed_response['Categories']['Category']
+
+            else:
+                self.logger.error("Error in response from bluecoat API!.")
+                return -1
+
+        except Exception as e:
+            self.logger.exception(e)
+            return -1
+
+        return parsed_response
+
+    def submit_url(self, url, submit_comment, cat1, cat2, email, customTrackingID, confidence):
+
+        '''
+        Submit an url to bluecoat for the given categories (cat1, cat2) and returns the submissionId for further usage
+        :param url: URL to submit
+        :param submit_comment: Comment for submission
+        :param cat1: One of the predefined categories
+        :param cat2: (Optional) One of the predefined categories
+        :param email: Email address which receives the outcome of the url review
+        :param customTrackingID: custom tracking ID (maximum 32 characters)
+        :param confidence: integer value between 0-100 --> 0=no confidence; 100=compete confidence
+
+        :return: uniq submissionId provided by BlueCoat API (7 characters, numbers only)
+        '''
+
+        payload = {
+            'userID': self.user,
+            'url': url,
+            'comments': submit_comment,
+            'cat1': cat1,
+            'cat2': cat2,  # Optional
+            'email': email,  # Optional
+            'customTrackingID': customTrackingID  # Optional
+        }
+
+        headers = {
+            'content-type': 'application/x-www-form-urlencoded'
+        }
+
+        try:
+            parsed_response = None
+            response = requests.post(self.report_threat_url, data=payload, headers=headers, verify=False)
+            if response.status_code == 200:
+                parsed_response = xmltodict.parse(response._content)
+                self.logger.debug("Successfully parsed response from BlueCoat-Submit for URL [" + str(url) + "]:" + str(
+                    parsed_response))
+            else:
+                self.logger.error("Error in response from BlueCoat API")
+
+        except Exception as e:
+            self.logger.exception(e)
+            return None
+
+        if parsed_response and len(parsed_response) > 0:
+
+            good_case = False
+
+            if "Accepted" in parsed_response or "Accepted" in parsed_response['SubmissionResult']:
+                self.logger.debug("Accepted response for url'" + url + "' Response:" + str(parsed_response))
+
+                if 'SubmissionResult' in parsed_response and 'Accepted' in parsed_response['SubmissionResult']:
+                    self.logger.info(
+                        "BlueCoat Accepted response for url'" + url + "' SubmissionResult (uniq id)=" + str(
```

Comment:
@wagner-certat: This logging is only possible with the baseLib. (And the rest of it in this libary)